app/src/main/java/com/yacgroup/yacguide/database/tourbook/JsonExporter.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2023 Christian Sommer
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package com.yacgroup.yacguide.database.tourbook
+
+import android.content.ContentResolver
+import android.net.Uri
+import com.yacgroup.yacguide.database.Ascend
+import com.yacgroup.yacguide.database.DatabaseWrapper
+import org.json.JSONArray
+import org.json.JSONException
+import org.json.JSONObject
+import java.io.IOException
+
+class JsonExporter(
+    private val _db: DatabaseWrapper,
+    _contentResolver: ContentResolver): BaseExporter(_contentResolver) {
```

Comment:
Actually I'm not happy at all with the underscores in the constructor parameters. Even if you declare the parameter as `private` it is visible for the user during instantiation because you need to assign something to this parameter.

Reply:
Me neither. But having a visual indicator for private class members provides (at least for me) a higher value than having consistent parameter names.